### PR TITLE
Minor touch up for async docs

### PR DESCRIPTION
**Pull request by dstandish**
I explain in comments inline

### CODE REVIEW

File: docs/apache-airflow/concepts/deferring.rst [hunk 1]
```diff
@@ -55,7 +55,7 @@ Writing Deferrable Operators
 
 Writing a deferrable operator takes a bit more work. There are some main points to consider:
 
-* Your Operator must defer itself based on a Trigger. If there is a Trigger in core Airflow you can use, great; otherwise, you will have to write one.
```
Comment (dstandish) [inline docs/apache-airflow/concepts/deferring.rst:58]: "Based on" makes it sound like the deferral is in _response_ to a trigger, that in a sense the trigger comes before the deferral.  By using "with" instead, I think it's slightly clearer that that you must include a trigger instance in your deferral, i.e. as part of the deferral, i.e. within the deferral exception.

File: docs/apache-airflow/concepts/deferring.rst [hunk 2]
```diff
@@ -67,13 +67,13 @@ Triggering Deferral
 If you want to trigger deferral, at any place in your Operator you can call ``self.defer(trigger, method_name, kwargs, timeout)``, which will raise a special exception that Airflow will catch. The arguments are:
 
 * ``trigger``: An instance of a Trigger that you wish to defer on. It will be serialized into the database.
-* ``method_name``: The method name on your Operator you want Airflow to call when it resumes, other than ``execute``.
```
Comment (dstandish) [inline docs/apache-airflow/concepts/deferring.rst:70]: elsewhere it is made clear that you can resume to `execute`

File: docs/apache-airflow/concepts/deferring.rst [hunk 3]
```diff
@@ -67,13 +67,13 @@ Triggering Deferral
...
 * ``kwargs``: Additional keyword arguments to pass to the method when it is called. Optional, defaults to ``{}``.
 * ``timeout``: A timedelta that specifies a timeout after which this deferral will fail, and fail the task instance. Optional, defaults to ``None``, meaning no timeout.
 
 When you opt to defer, your Operator will *stop executing at that point and be removed from its current worker*. No state - such as local variables, or attributes set on ``self`` - will persist, and when your Operator is resumed it will be a *brand new instance* of it. The only way you can pass state from the old instance of the Operator to the new one is via ``method_name`` and ``kwargs``.
 
-When your Operator is resumed, you will find an ``event`` item added to the kwargs passed to it, which contains the payload from the trigger event that resumed your Operator. Depending on the trigger, this may be useful to your operator (e.g. it's a status code or URL to fetch results), or it may not be important (it's just a datetime). Your ``method_name`` method, however, *must* accept ``event`` as a keyword argument.
```
Comment (dstandish) [inline docs/apache-airflow/concepts/deferring.rst:76]: to `it` sounds like to `the operator` but it's to `the method`

File: docs/apache-airflow/concepts/deferring.rst [hunk 4]
```diff
@@ -140,8 +145,6 @@ Triggers can be as complex or as simple as you like provided you keep inside thi
 
 If you are new to writing asynchronous Python, you should be very careful writing your ``run()`` method; Python's async model means that any code that does not correctly ``await`` when it does a blocking operation will block the *entire process*. Airflow will attempt to detect this and warn you in the triggerer logs when it happens, but we strongly suggest you set the variable ``PYTHONASYNCIODEBUG=1`` when you are writing your Trigger to enable extra checks from Python to make sure you're writing non-blocking code. Be especially careful when doing filesystem calls, as if the underlying filesystem is network-backed it may be blocking.
 
-Right now, Triggers are only used up to their first event, as they are only used for resuming deferred tasks (which happens on the first event fired). However, we plan to allow DAGs to be launched from triggers in future, which is where multi-event triggers will be more useful.
```
Comment (dstandish) [inline docs/apache-airflow/concepts/deferring.rst:143]: i moved this to a `note` immediately after where the notion of multi-event triggers is introduced.  until i found this paragraph, i thought multi-event triggers were already supported and was confused when i tried to understand how they could be used.

**Review by kaxil — APPROVED** [review]

**Pull request merged**